### ISSUE TITLE
Support for SNS RawMessageDelivery

### ISSUE THREAD

**Issue by robwettach** [aws/aws-cdk#1242]
In CloudFormation, SNS's Subscription resource supports the "RawMessageDelivery" flag ([docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-subscription.html#cfn-sns-subscription-rawmessagedelivery)), but the CDK's `Subscription` Construct does not ([SubscriptionProps](https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-sns/lib/subscription.ts#L8)).  My existing app built with CloudFormation uses the "RawMessageDelivery" flag, so I'd prefer not to have to modify my application code to parse the SNS event for my message body.

**Comment by eladb** [comment]
As an interim work around you should be able to use the escape hatch: https://awslabs.github.io/aws-cdk/aws-construct-lib.html#access-the-aws-cloudformation-layer

**Comment by robwettach** [comment]
Thanks.  I was able to get `RawMessageDelivery` to show up in my output CloudFormation by using the `addOverride` method on the inner `SubscriptionResource`.

I'm using the Java CDK bindings and ran into an issue using the `getPropertyOverrides()` method where I'd get a stack overflow when running `cdk synth`.  I've written up a minimal example here: https://github.com/robwettach/cdk-1242-example .  I also created the same stack in TypeScript and didn't run into this problem, so there might be something going on with the Jsii layer.
